app/assets/javascripts/google_maps/custom_marker.js

```diff
@@ -0,0 +1,55 @@
+// Utility function to build custom html markers
+//
+function CustomMarker(latlng, map, args) {
+  this.latlng = latlng;	
+  this.args = args;	
+  this.setMap(map);	
+}
+
+CustomMarker.prototype = new google.maps.OverlayView();
+
+CustomMarker.prototype.draw = function() {
+
+  var self = this;
+
+  var div = this.div;
+
+  if (!div) {
+
+    div = this.div = document.createElement('div');
+
+    div.className = 'custom-marker';
+
+    div.style.position = 'absolute';
+    div.style.cursor = 'pointer';
+    div.innerHTML = self.args.content;
+
+
+    google.maps.event.addDomListener(div, "click", function(event) {			
+      google.maps.event.trigger(self, "click");
+    });
+
+    var panes = this.getPanes();
+    panes.overlayImage.appendChild(div);
+  }
+
+  var point = this.getProjection().fromLatLngToDivPixel(this.latlng);
+
+  // By default the the top left corner of custom marker is used to set marker position
+  // check the explancation from: http://humaan.com/custom-html-markers-google-maps/
+  if (point) {
+    div.style.left = (point.x - ($('.custom-marker').width() / 2)) + 'px';
+    div.style.top = (point.y - $('.custom-marker').height()) + 'px';
```

Comment:
Do these markers have a consistent height and width? If so, I wonder if we could avoid doing this calculation for each marker?